src/main/java/org/openmaptiles/layers/WaterName.java

```diff
@@ -176,20 +189,32 @@ public void process(Tables.OsmWaterPolygon element, FeatureCollector features) {
         Geometry centerlineGeometry = lakeCenterlines.get(element.source().id());
         FeatureCollector.Feature feature;
         int minzoom = 9;
+        String place = element.place();
+        String clazz;
+        if ("bay".equals(element.natural())) {
+          clazz = FieldValues.CLASS_BAY;
+        } else if ("sea".equals(place)) {
+          clazz = FieldValues.CLASS_SEA;
+        } else {
+          clazz = FieldValues.CLASS_LAKE;
+          minzoom = 3;
+        }
         if (centerlineGeometry != null) {
           // prefer lake centerline if it exists
           feature = features.geometry(LAYER_NAME, centerlineGeometry)
             .setMinPixelSizeBelowZoom(13, 6d * element.name().length());
         } else {
           // otherwise just use a label point inside the lake
           feature = features.pointOnSurface(LAYER_NAME);
-          Geometry geometry = element.source().worldGeometry();
-          double area = geometry.getArea();
-          minzoom = (int) Math.floor(20 - Math.log(area / WORLD_AREA_FOR_70K_SQUARE_METERS) / LOG2);
-          minzoom = Math.min(14, Math.max(9, minzoom));
+          double area = element.source().area();
+          if (place != null && SEA_OR_OCEAN_PLACE.contains(place)) {
+            minzoom = 0;
```

Comment:
Based on this logic:

https://github.com/openmaptiles/openmaptiles/compare/v3.14..master#diff-201daa1c61c99073fe3280d440c9feca5ed2236b251ad454caa14cc203f952d1R54-R56

Don't we want minzoom to be set by area for sea or ocean as well?

Reply:
Yes, we do. I've botched following: https://github.com/openmaptiles/openmaptiles/pull/1457/files#diff-201daa1c61c99073fe3280d440c9feca5ed2236b251ad454caa14cc203f952d1R55

Thanks, adjusted.